lib/util/common.js

```diff
@@ -93,6 +94,10 @@ function shouldRouteHide (schema, opts) {
     return schema.tags.includes(hiddenTag)
   }
 
+  if (apiVersion && routeVersion && !semverSatisfies(routeVersion, apiVersion)) {
+    return true
+  }
```

Comment:
By looking at this now, I think we could implemented a slightly more generic implementation that allows the user to provide a function instead, and we could add an example for that case.

Reply:
Do you mean something like this?

```js
fastify.register(fastifySwagger, {
  filterRoutes: (route) => {
    // some custom logic
    return true;
  }
})
```

So for my use case it would be

```js
fastify.register(fastifySwagger, {
  filterRoutes: (route) => {
    if(route?.constraints?.version) {
      return semverSatisfies(route?.constraints?.version,'^1.0.0')
    }
    return true
  }
})
```